extension.go

```diff
@@ -89,6 +91,7 @@ func (ctx CompilerContext) GetResourceSchema() *Schema {
 
 // ValidationContext provides additional context required in validating for extension.
 type ValidationContext struct {
+	ctx             context.Context
```

Comment:
We will need to add a public method to ValidationContext that returns the new `.ctx`.
We will also need to think about what we want to call this context, as it is confusing to simply call it "context" and place it in something else that's called ValidationContext.

I think of this as "extra context". If we agree on naming, then I would:
- Name the new public Validate method `ValidateWithExtraContext`
- Name the new private property `extraContext`
- Add new method on `ValidationContext` that returns `.extraContext` - call that method `GetExtraContext`